api/document/js/serialize-doc.ts

```diff
@@ -41,8 +27,11 @@ export const apiFactory = {
   }),
 };
 
+type NodeConverterMap = {
+  [nodeName: string]: (node: Node) => ApiNode,
```

Comment:
In the future, I'd like to improve upon this to actually require that `nodeName` be a valid node type in our schema.  That said, I might be venturing into "type astronaut" territory with that one...

Reply:
I think that meshes with your earlier concerns around logic living in lots of different places. I can imagine we'd have an interface like
```typescript
interface SchemaNode {
  ...
  fromApi: (node: ApiNode) => ProseMirror.Node?,
  toApi: () => ApiNode,
  deleteWhenEmpty: boolean,
  ...
}
```
at some point.